darshan-hpc/darshan
# MAINT: refactor assets

Supports the testing of gh-597 on HPC installs with `pytest --pyargs darshan`
from any directory (HPC users should not have to clone `darshan` to test
an already-installed HPC version).

* move `examples` and `tests` folders inside
`darshan` package so they can be easily
installed as elements of the package, and prevent us
from having to search more broadly on the filesystem
to locate `*.darshan` file assets (and to avoid monopolizing
`import tests`, `import examples`, which is crazy)

* adjust setup.py accordingly--there is now only a
`darshan` package to `install`, not separate `tests`
and `examples` packages

* migrate the test suite over to `get_log_path()` as needed
to empower a portable approach to running tests (i.e.,
`pytest --pyargs darshan` from any directory)

* `_locate_log()` no longer needs to go searching around
on the filesystem outside of our own packages, so get
rid of that..

* on this branch, in some random directory,
`pytest --pyargs darshan -n 8` produces:
`255 passed in 88.93s (0:01:28)`

* this is in line with latest `pydarshan-devel` run
of the full suite before migration to `get_log_path()`
and reorganization:
`pytest --import-mode=importlib -n 8`
`255 passed in 87.78s (0:01:27)`

* enforce support for `pytest --pyargs darshan` from any
dir in the CI (using a random temp dir there)

* things reviewers could do to help here:
  - check the large diff for potential errors
  - check/fix up the `asv` benchmark suite for the
    effects of this branch
  - check if the coverage report stuff is still working
    "ok" with new `pytest` portability/incantation
